packages/wrangler/src/dev/miniflare.ts

```diff
@@ -136,25 +143,18 @@ function createNotFoundWorkerEntrypointClass({ service }) {
 
 	return klass;
 }
+`;
 
-export default {
-	async fetch(request, env) {
-		const originalUrl = request.headers.get(HEADER_URL);
-		const className = request.headers.get(HEADER_NAME);
-		const idString = request.headers.get(HEADER_ID);
-		const cf = JSON.parse(request.headers.get(HEADER_CF_BLOB));
-		if (originalUrl === null || className === null || idString === null) {
-			return new Response("[wrangler] Received Durable Object proxy request with missing headers", { status: 400 });
-		}
-		request = new Request(originalUrl, request);
-		request.headers.delete(HEADER_URL);
-		request.headers.delete(HEADER_NAME);
-		request.headers.delete(HEADER_ID);
-		request.headers.delete(HEADER_CF_BLOB);
-		const ns = env[className];
-		const id = ns.idFromString(idString);
-		const stub = ns.get(id);
-		return stub.fetch(request, { cf });
+const EXTERNAL_SERVICE_RECEIVER_NAME =
+	"__Wrangler__LocalOnly__ExternalDurableObjectsReceiver";
+const EXTERNAL_SERVICE_RECEIVER_SCRIPT = `
+\n;; ${/* this script needs to be concatenated with the user's script :( -- use this to avoid ASI syntax errors*/ ""}
+import { WorkerEntrypoint as __Wrangler__LocalOnly__WorkerEntrypoint } from "cloudflare:workers";
+export class ${EXTERNAL_SERVICE_RECEIVER_NAME} extends __Wrangler__LocalOnly__WorkerEntrypoint {
+    async proxyMethod({ bindingName, doId, method, args }) {
+        const ns = this.env[bindingName];
+        const stub = ns.get(ns.idFromString(doId));
+        return stub[method](...args);
 	}
 }
 `;
```

Comment:
This script is added to the user's script – not a separate service as the previous implementation did – so we can bind to it as a regular external entrypoint

This may sound scary but I've taken care to use long+clear names to avoid clashing with user variables in scope and that explain their purpose so if a user ever does discover them they will understand its: (a) injected by wrangler and (b) in local code only, not deployed code